bpf/process/pfilter.h

```diff
@@ -114,6 +114,141 @@ struct selector_filter {
 	__u32 index;
 };
 
+enum {
+	PF_CRED_UNKNOWN = 0x00, // No filter
+	PF_CRED_RUID = 0x01, // Real UIDs
+	PF_CRED_RGID = 0x02, // Real GIDs
+	PF_CRED_EUID = 0x04, // Effective UIDs
+	PF_CRED_EGID = 0x08, // Effective GIDs
+};
+
+struct cred_id {
+	__u32 start;
+	__u32 end;
+} __attribute__((packed));
+
+struct cred_filter {
+	__u64 ty; /* credentials type flags */
+	__u32 op; /* operator */
+	__u32 num; /* number of values */
+	struct cred_id val[];
+} __attribute__((packed));
+
+#define MAX_CRED_SELECTOR_FIELDS 2
+#define MAX_CRED_SELECTOR_VALUES 3
+
+#ifdef __LARGE_BPF_PROG
+
+FUNC_INLINE int
+__match_cred_uids(__u32 *f, __u32 *index, struct cred_filter *cred_f, __u32 uid)
+{
+	__u32 min, max;
+
+	min = *(__u32 *)((__u64)f + (*index & INDEX_MASK));
+	max = *(__u32 *)((__u64)f + ((*index + 4) & INDEX_MASK));
+	if (cred_f->op == op_filter_eq && (uid >= min && uid <= max))
+		return PFILTER_ACCEPT;
+	else if (cred_f->op == op_filter_neq && (uid < min || uid > max))
+		return PFILTER_ACCEPT;
+
+	return PFILTER_REJECT;
+}
+
+FUNC_INLINE int
+match_cred_uids(__u32 *f, __u32 *index, struct cred_filter *cred_f, __u32 uid)
+{
+	int i, ret = 0;
+
+	if (!cred_f->ty)
+		return PFILTER_REJECT;
+
+	if (cred_f->op == op_filter_neq)
+		ret = 1;
+
+	for (i = 0; i < cred_f->num; i++) {
+		if (i > (MAX_CRED_SELECTOR_VALUES - 1)) // pass the verifier
+			break;
+
+		if (cred_f->op == op_filter_eq) {
+			ret |= __match_cred_uids(f, index, cred_f, uid);
+			if (ret == PFILTER_ACCEPT) {
+				/* break loop and return */
+				/* now index points at the end of cred_filter filter */
+				*index += sizeof(struct cred_id) * (cred_f->num - i);
+				return ret;
+			}
+		} else {
+			ret &= __match_cred_uids(f, index, cred_f, uid);
+			if (ret == PFILTER_REJECT) {
+				/* break loop and return */
+				/* now index points at the end of cred_filter filter */
+				*index += sizeof(struct cred_id) * (cred_f->num - i);
+				return ret;
+			}
+		}
+		*index += sizeof(struct cred_id);
+		/* now index points at the end of cred_filter filter */
+	}
+
+	return ret;
+}
+
+FUNC_INLINE int
+process_filter_current_cred(__u32 *f, __u32 *index, __u32 len, struct msg_generic_kprobe *msg)
+{
+	__u32 flags, ret = 1, uid;
```

Comment:
let's init ret with PFILTER_ACCEPT?

Reply:
I ended up with PFILTER_REJECT and adapted the code below